fortran-language-server: init at 1.12.0

sheepforce opened the pull request:
###### Motivation for this change
Adds the Fortran-Language-Server, which adds editor integration for Fortran developers.

###### Things done
- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

SuperSandro2000 commented on pkgs/development/tools/fortran-language-server/default.nix:
```diff
@@ -0,0 +1,18 @@
+    inherit pname version;
+    sha256 = "7Dkh7yPX4rULkzfJFxg47YxrCaxuHk+k3TOINHS9T5A=";
+  };
+
```
Please add a checkPhase and add a pythonImportsCheck

sheepforce replied:
This is a pure executable Python application. There is no documented module one could import and there is also no test case, the fls isn't meant to be used interactively from a command line. Which kind of test would you suggest? I am a little bit lost there.

SuperSandro2000 replied:
You should be able to import it https://github.com/hansec/fortran-language-server/blob/master/fortls.py#L7

Does the cli have a --help or something?

sheepforce replied:
Ah nice, I've added the import check and a `fortls --help` invocation and both work fine. Thanks for the ideas :)

sheepforce committed:
fortran-language-server: init at 1.12.0


fortran-language-server: add pythonImportsCheck and a `--help`cli interaction


fortran-language-server: do not discard errors
